dcos-metrics.go

```diff
@@ -58,7 +57,12 @@ func main() {
 	if producerIsConfigured("http", cfg) {
 		log.Info("HTTP producer enabled")
 		cfg.Producers.HTTPProducerConfig.DCOSRole = cfg.DCOSRole
-		cfg.Producers.HTTPProducerConfig.CacheExpiry = time.Duration(cfg.Collector.MesosAgent.PollPeriod) * time.Minute * 2
+
+		minCacheExpiry := cfg.Collector.MesosAgent.PollPeriod * 2
+		if cfg.Producers.HTTPProducerConfig.CacheExpiry < minCacheExpiry {
```

Comment:
I think the [original code](https://github.com/dcos/dcos-metrics/blob/master/dcos-metrics.go#L61) exists to prevent metrics being removed from the cache prematurely (eg if PollPeriod was 1m and CacheExpiry was 30s, metrics would not always be available for a long-running service). 

This was also kind of crappy because a longer cache expiry in the configuration file would be overridden silently. 

We now fix the erroneous multiplication by `time.Minute`, honour the user's configuration if it's sensible and warn them if it's not.